src/sinks/elasticsearch/common.rs

```diff
@@ -122,6 +123,59 @@ impl ElasticsearchCommon {
 
         let region = config.aws.as_ref().and_then(|config| config.region());
 
+        let version = if let Some(version) = *version {
+            version
+        } else {
+            let ver = match config.api_version {
+                ElasticsearchApiVersion::V6 => 6,
+                ElasticsearchApiVersion::V7 => 7,
+                ElasticsearchApiVersion::V8 => 8,
+                ElasticsearchApiVersion::Auto => {
+                    match get_version(
+                        &base_url,
+                        &http_auth,
+                        &aws_auth,
+                        &region,
+                        &request,
+                        &tls_settings,
+                        proxy_config,
+                    )
+                    .await
+                    {
+                        Ok(version) => version,
+                        // This error should be fatal, but for now we only emit it as a warning
+                        // to make the transition smoother.
+                        Err(error) => {
+                            warn!(message = "Failed to determine Elasticsearch version. Please set option `api_version`.", %error);
+                            // For now, estimate version.
+                            match config.suppress_type_name {
+                                Some(true) => 8,
+                                _ => 6,
+                            }
+                        }
```

Comment:
I might include the API version Vector decided to use in the warning message. Also could say something a bit more verbose like:

```
Failed to determine Elasticsearch version from `/_cluster/state/version`. Please fix the reported error or set an API version explicitly via `api_version`.
```

I'd also set the default to `7` as that will be less confusing for users since `6` is undocumented.

Reply:
Setting default as 7 would break configs since we `suppress_type_name` for versions 7 and greater. We can change this so that we suppress type name for 8 and greater, but then for version 7 we would receive warnings from Elasticsearch server.

Reply:
Ah, yeah I see 🤔 We may want to publicly document support for 6 then and also add it to our integration test matrix to make sure we continue to maintain support.

Reply:
v6 is now documented.

Regarding integration test matrix, from what I see v7 is the only one tested at the moment as defined at https://github.com/vectordotdev/vector/blob/067795eabedea44db8d5d069bc8db1a4646dcc9d/scripts/integration/docker-compose.elasticsearch.yml#L11

So I'm guessing a new github workflow similar to [kubernetes one](.github/workflows/k8s_e2e.yml) needs to be added to be able to test multiple versions, v6, v7, and v8.

Reply:
Thanks for adding the v6 documentation!

I opened https://github.com/vectordotdev/vector/issues/15163 , to track improving the testing coverage for other versions.